src/Microsoft.AspNetCore.Mvc.Core/Internal/ObjectResultExecutor.cs

```diff
@@ -175,71 +181,57 @@ public virtual Task ExecuteAsync(ActionContext context, ObjectResult result)
                 throw new ArgumentNullException(nameof(formatters));
             }
 
-            // Check if any content-type was explicitly set (for example, via ProducesAttribute
-            // or URL path extension mapping). If yes, then ignore content-negotiation and use this content-type.
-            if (contentTypes.Count == 1)
+            var request = formatterContext.HttpContext.Request;
+            var acceptableMediaTypes = GetAcceptableMediaTypes(contentTypes, request);
+            var selectedFormatter = default(IOutputFormatter);
```

Comment:
FYI only: The main downside of this is some developers read `default` and think the type is a `struct`. In this case, they then have to _think_ about `struct`s and `interface`s :smiling_imp: 

In future I suggest sticking w/ `IOutputFormatter selectedFormatter = null;`, an approach that also works when the variable doesn't need to be initialized.